### PR TITLE
Implement protobuf linting, formatting, and breaking change detection

### PR DESCRIPTION
**Pull request by jhoward-lm**
This PR implements configuration and automation for Buf CLI, which provides linting, formatting, and breaking change detection for protocol buffer definitions.

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
lgtm


i am ok to add a taskfile, will be good to have some kind of documentation on that (non blocking to merge)

thanks for doing this

**Comment by jhoward-lm** [comment]
> lgtm
> 
> i am ok to add a taskfile, will be good to have some kind of documentation on that (non blocking to merge)
> 
> thanks for doing this

@cpanato I added some instructions in the `docs/regenerate-protobuf.md` locally, waiting to hear back from @puerco to see if he wants me to restore the original order of message/enum definitions to make the diff cleaner. Trying to minimize the number of pushes because I believe it will wipe out approvals every time I do.

File: Makefile [hunk 1]
```diff
@@ -16,11 +18,21 @@ conformance: ## Regenerate the conformance golden samples. Only to be run when c
...
+buf-format: ## Check for breaking changes in protocol buffer definitions
+	buf format --write
+
+.PHONY: buf-lint
+buf-lint: ## Check for breaking changes in protocol buffer definitions
+	buf format --diff --exit-code
+	buf lint
+	buf breaking --against .git#tag=${VERSION},subdir=api
+
+.PHONY: buf-generate
```
[the remark this answers is not in the document]
Comment (jhoward-lm) [reply Makefile]: I reverted it

**Review by puerco — APPROVED** [review]
Wonderful, thanks for working on this, great work! :partying_face:

**Pull request merged**